includes/handlers/class-webmention-handler-meta.php

```diff
@@ -0,0 +1,224 @@
+<?php
+/**
+ * Class for webmention parsing using META tags.
+ */
+class Webmention_Handler_Meta extends Webmention_Handler_Base {
+
+	/**
+	 * Handler Slug to Uniquely Identify it.
+	 *
+	 * @var string
+	 */
+	protected $slug = 'meta';
+
+	/**
+	 * Takes a request object and parses it.
+	 *
+	 * @param Webmention_Request $request Request Object.
+	 * @param Webmention_Item $item A Parsed Item. If null, a new one will be created.
+	 * @return WP_Error|true Return error or true if successful.
+	 */
+	public function parse( $request, $item = null ) {
+		if ( $item instanceof Webmention_Item ) {
+			$this->webmention_item = $item;
+		} else {
+			$this->webmention_item = new Webmention_Item();
+		}
+		$dom  = clone $request->get_domdocument();
+		$meta = $this->parse_meta( $dom );
+
+		// Set raw data.
+		$this->webmention_item->set__raw( $meta );
+
+		// OGP has no concept of anything but mention so it is always a mention.
+		$this->webmention_item->set__response_type = 'mention';
+
+		$this->ogp_to_jf2( $meta );
+		$this->dublin_core_to_jf2( $meta );
+
+		// If Site Name is not set use domain name less www
+		if ( ! $this->webmention_item->has_site_name() && $this->webmention_item->has_url() ) {
+			$this->webmention_item->set__site_name( preg_replace( '/^www\./', '', wp_parse_url( $this->webmention_item->get_url(), PHP_URL_HOST ) ) );
+		}
+
+		$this->webmention_item->set_name( $meta['title'] );
+
+		if ( ! $this->webmention_item->has_published() ) {
+			if ( isset( $meta['citation_date'] ) ) {
```

Comment:
this could be also part of the setter, so we do not have to check this every time.